witness/src/Dscp/Witness/Logic/Getters.hs

```diff
@@ -112,26 +125,18 @@ resolvePrevious o = do
 -- | Given the element, get the next one.
 resolveNext
     :: (HasWitnessConfig, HasHeaderHash x)
-    => x -> SdM_ chgacc (Maybe HeaderHash)
+    => x -> SdM (Maybe HeaderHash)
 resolveNext = SD.queryOne . NextBlockOf . headerHash >=> pure . map unNextBlock
 
 ----------------------------------------------------------------------------
 -- Account getters
 ----------------------------------------------------------------------------
 
--- TODO [DSCP-367] Unite these two below
--- | Safely get an account.
-getAccountMaybe
-    :: WitnessWorkMode ctx m
-    => Address -> SdReadM 'ChainOnly m (Maybe Account)
-getAccountMaybe =
-    lift . runStateSdMLocked . SD.queryOne . AccountId
-
 -- | Safely get an account taking mempool into consideration.
-getMempoolAccountMaybe
-    :: WitnessWorkMode ctx m
-    => Address -> SdReadM 'ChainAndMempool m (Maybe Account)
-getMempoolAccountMaybe addr = readingSDLock $ liftSdM $ SD.queryOne (AccountId addr)
+getAccountMaybe
+    :: (WitnessWorkMode ctx m, KnownSdReadMode mode, WithinReadSDLock)
+    => Address -> SdReadM mode m (Maybe Account)
+getAccountMaybe addr = liftSdM $ SD.queryOne (AccountId addr)
```

Comment:
So, there were "get account' and "get account from mempool", both doing `SD.queryOne`. What does the survivor do?

Reply:
Asks for an account in the mode requested by the caller of this method.